employee-rostering-gwtui/src/test/java/org/optaweb/employeerostering/gwtui/client/pages/shiftroster/ShiftEditFormTest.java

```diff
@@ -0,0 +1,387 @@
+        doReturn(spotList.get(2)).when(mockShiftGridObject).getSpot();
+
+        restCallbackAnswers.add(spotList);
+        restCallbackAnswers.add(employeeList);
+        testedShiftEditForm.init(mockShiftGridObject);
+
+        doReturn(spotList.get(3).getId().toString()).when(spotSelect).getSelectedValue();
+        doReturn(employeeList.get(1).getId().toString()).when(employeeSelect).getSelectedValue();
+        doReturn(LocalDateTime.of(2012, 10, 3, 9, 0)).when(from).getValue();
+        doReturn(LocalDateTime.of(2012, 10, 3, 14, 0)).when(to).getValue();
+
+        restCallbackAnswers.add(shiftView);
+
+        MouseEvent mouseEvent = mock(MouseEvent.class);
+        testedShiftEditForm.onApplyButtonClick(mouseEvent);
+        verify(testedShiftEditForm).updateShiftFromWidgetsIfValid(shiftView);
+        verify(testedShiftEditForm).hide();
+        verify(eventManager).fireEvent(EventManager.Event.SHIFT_ROSTER_INVALIDATE);
+    }
+
+    @Test
+    public void testInvalidUpdateShiftFromForm() {
+        ShiftView shiftView = new ShiftView(0, spotList.get(2),
+                                            LocalDateTime.of(2012, 10, 1, 9, 0),
+                                            LocalDateTime.of(2012, 10, 1, 14, 0));
+        doReturn(shiftView).when(mockShiftGridObject).getShiftView();
+        doReturn(spotList.get(2)).when(mockShiftGridObject).getSpot();
+
+        restCallbackAnswers.add(spotList);
+        restCallbackAnswers.add(employeeList);
+        testedShiftEditForm.init(mockShiftGridObject);
+
+        doReturn(spotList.get(3).getId().toString()).when(spotSelect).getSelectedValue();
+        doReturn(employeeList.get(1).getId().toString()).when(employeeSelect).getSelectedValue();
+        doReturn(LocalDateTime.of(2012, 10, 3, 14, 0)).when(from).getValue();
+        doReturn(LocalDateTime.of(2012, 10, 3, 9, 0)).when(to).getValue();
+
+        restCallbackAnswers.add(shiftView);
+
+        MouseEvent mouseEvent = mock(MouseEvent.class);
+        testedShiftEditForm.onApplyButtonClick(mouseEvent);
+        verify(testedShiftEditForm).updateShiftFromWidgetsIfValid(shiftView);
+        verify(eventManager, VerificationModeFactory.noMoreInteractions()).fireEvent(any());
+    }
+
+    @Test
+    public void testDelete() {
+        doReturn(mockLane).when(mockShiftGridObject).getLane();
+
+        ShiftView shiftView = new ShiftView(0, spotList.get(2),
+                                            LocalDateTime.of(2012, 10, 1, 9, 0),
+                                            LocalDateTime.of(2012, 10, 1, 14, 0));
+        doReturn(shiftView).when(mockShiftGridObject).getShiftView();
+        doReturn(spotList.get(2)).when(mockShiftGridObject).getSpot();
+
+        restCallbackAnswers.add(spotList);
+        restCallbackAnswers.add(employeeList);
+        testedShiftEditForm.init(mockShiftGridObject);
+
+        MouseEvent mouseEvent = mock(MouseEvent.class);
+        restCallbackAnswers.add(null);
```

Comment:
Adding a null into a collection is very unusual. If there is not other way to do it, could you please at least put there a comment explaining the motivation?

Reply:
Fixed; changed it into two different test cases; one for a successful delete (true) and another for a failed delete (false).